Client/Application/AppDelegate.swift

```diff
@@ -865,6 +871,15 @@ extension AppDelegate {
         // For now, we should just re-process the message handling.
         let handler = FxAPushMessageHandler(with: profile)
         handler.handle(userInfo: userInfo).upon { res in
+            if let message = res.successValue {
+                switch message {
+                case .thisDeviceDisconnected:
+                    FxALoginHelper.sharedInstance.applicationDidDisconnect(application)
```

Comment:
I've posted the message on the `DispatchQueue.main` in da8c236, rather than forcing the whole of the disconnect code to be run on the UI thread.